src/trust_region.jl

```diff
@@ -143,18 +143,18 @@ function trust_region_{T}(df::AbstractDifferentiableMultivariateFunction,
         dogleg!(p, r, d, J, delta)
 
         copy!(xold, x)
-        Base.BLAS.axpy!(one(T), p, x)  # x += p
+        x .+= p
         df.f!(x, r_new)
         f_calls += 1
 
         # Ratio of actual to predicted reduction (equation 11.47 in N&W)
         A_mul_B!(r_predict, J, p)
-        Base.BLAS.axpy!(one(T), r, r_predict)
+        r_predict .+= r
         rho = (sum(abs2,r) - sum(abs2,r_new)) / (sum(abs2,r) - sum(abs2,r_predict))
 
         if rho > eta
             # Successful iteration
-            copy!(r, r_new)
```

Comment:
I was thinking performance .

If they are the same then I'd id say the broadcast one is clearer.

Let's merge